### PR TITLE
Created wrap definitions for 2.3.1.

### PR DESCRIPTION
**Pull request by jpakkane**

### CODE REVIEW

**Comment by igor-raits** [comment]
- [x] if new project, master has tagged commit as only commit
- [x] if new branch, it is branched from master
- [x] contains a readme.txt
- [x] contains an upstream.wrap file
- [x] download link points to authoritative upstream location
- [x] wrap repository contains only build system files
- [x] merge request is pointed to correct target branch (not master)
- [x] wrap works
- [x] repo does not have useless top level directory (i.e. libfoobar-1.0.0)

File: Box2D/HelloWorld/meson.build [hunk 1]
```diff
@@ -0,0 +1,3 @@
+executable('HelloWorld', 'HelloWorld.cpp',
+  include_directories : inc,
+  link_with : box2dlib)
```
Comment (igor-raits) [inline Box2D/HelloWorld/meson.build:3]: Shouldn't we have `test()` for it?

File: upstream.wrap [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[wrap-file]
+subdir = box2d-2.3.1
+
+source_url = https://codeload.github.com/erincatto/box2d/zip/v2.3.1
```
Comment (igor-raits) [inline upstream.wrap]: I'd preffer to use https://github.com/erincatto/box2d/archive/v2.3.1/box2d-2.3.1.tar.gz

What do you think?

Comment (jpakkane) [inline upstream.wrap]: That is just a redirect to the codeload address.

File: upstream.wrap [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+source_filename = box2d_2.3.1.zip
+source_hash = 498b0a32cef46c5580e53a7ad5aa002926c036379a1c41767b662ae99c7c32b3
+
```
Comment (igor-raits) [inline upstream.wrap]: please no blank line at EOF. wrapdb generates blank line here

File: upstream.wrap (hunk 2 above)
Comment (igor-raits) [inline upstream.wrap]: and it sets properly filename

Comment (igor-raits) [inline upstream.wrap]:
```
Length: 2864860 (2.7M) [application/zip]
Saving to: ‘v2.3.1’
```

vs

```
Length: 2600384 (2.5M) [application/x-gzip]
Saving to: ‘box2d-2.3.1.tar.gz’
```

File: meson.build [hunk 4]
```diff
@@ -0,0 +1,5 @@
+project('box2d', 'cpp')
+
+inc = include_directories('Box2D')
```
Comment (igor-raits) [inline meson.build]: just style: let's use `incdir` as in other projects

File: Box2D/Box2D/meson.build [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+ 'Dynamics/Joints/b2Joint.cpp',
+ 'Dynamics/Joints/b2MotorJoint.cpp',
+ 'Dynamics/Joints/b2MouseJoint.cpp',
+ 'Dynamics/Joints/b2PrismaticJoint.cpp',
+ 'Dynamics/Joints/b2PulleyJoint.cpp',
+ 'Dynamics/Joints/b2RevoluteJoint.cpp',
+ 'Dynamics/Joints/b2RopeJoint.cpp',
+ 'Dynamics/Joints/b2WeldJoint.cpp',
+ 'Dynamics/Joints/b2WheelJoint.cpp',
+ 'Rope/b2Rope.cpp',
```
Comment (igor-raits) [inline Box2D/Box2D/meson.build]: just style: let's use 2 spaces as in our meson.build rules for unittests ;)

**Pull request merged**